mail/api.py

```diff
@@ -133,10 +135,18 @@ def send_batch(cls, subject, body, recipients,  # pylint: disable=too-many-argum
             chunk_dict = {email: context for email, context in chunk}
             emails = list(chunk_dict.keys())
 
+            # parse our HTML body in order to generate a plain-text fallback
+            # the only thing we need to do manually is ensure that we keep the
+            # href for any URLs in the text
+            soup = BeautifulSoup(body, 'html5lib')
+            for link in soup.find_all('a'):
+                link.replace_with(link.attrs['href'])
```

Comment:
I just realized there's no reason to recalculate this every time we loop through the recipients, so I'm going to move this outside of the loop.